src/Compilers/Core/Portable/Compilation/IExpression.cs

```diff
@@ -841,10 +841,36 @@ public interface IConversionExpression : IHasOperatorMethodExpression
         /// Kind of conversion.
         /// </summary>
         ConversionKind ConversionKind { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an identity conversion.
+        /// </summary>
+        bool IsIdentity { get; }
+
+        /// <summary>
+        /// Returns true if the conversion is an implicit ('widening' in VB) conversion.
+        /// </summary>
+        bool IsImplicit { get; }
```

Comment:
That's why i asked about " Is there a difference between 'IsExplicit' and 'ExplicitCastInCode'?"

It sounds like there is.  If so, i recommend renaming IsExplicit to something more like "IsCast" (or "HasSourceCast"), and then having an actual IsExplicit that maps to IsExplicit (or 'IsNarrowing').